### PR TITLE
Prepare 2.10.4 for release

### PR DESCRIPTION
**Pull request by rpeden**
This PR adds release notes and updates the announcement banner for Prefect 2.10.4.

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+Prior to this release, removing keys from a project's `deployment.yaml` caused an error. Thanks to the changes in [#9190](https://github.com/PrefectHQ/prefect/pull/9190), Prefect now uses default values for any required keys missing from 
+your project's configuration.
+
+### Enhancements
+- Allow partial deployment.yaml files for projects by using defaults for missing values — [#9190](https://github.com/PrefectHQ/prefect/pull/9190)
+- Add flow run cancellation support for workers - [#9198](https://github.com/PrefectHQ/prefect/pull/9198)
+
+### Fixes
+- Prevent scheduled flow runs from getting stuck in `CANCELLING` state  — [#8414](https://github.com/PrefectHQ/prefect/pull/8414)
+- Fix `work_queues` and `worker_type` arguments for the `prefect worker start` CLI command — [#9154](https://github.com/PrefectHQ/prefect/pull/9154)
```
Comment (pleek91) [inline RELEASE-NOTES.md:18]:
```suggestion
- Fix `work_queues` and `worker_type` arguments for the `prefect worker start` CLI command — [#9154](https://github.com/PrefectHQ/prefect/pull/9154)
- Logs overflow container in ui [`#1342`](https://github.com/PrefectHQ/prefect-ui-library/pull/1342)
- Schema form incorrectly marking reference objects as optional [`#1332`](https://github.com/PrefectHQ/prefect-ui-library/pull/1332)
- Prevent flow graph shortcuts when metakey [`#1333`](https://github.com/PrefectHQ/prefect-ui-library/pull/1333)
```

File: docs/overrides/main.html [hunk 2]
```diff
@@ -2,10 +2,10 @@
 
 {% block announce %}
 <div style="color: white;">
-  <a href="https://github.com/PrefectHQ/prefect/blob/main/RELEASE-NOTES.md#release-2103" style="color: #07e798;">
-    <strong>Prefect 2.10.3 is here!</strong>
-  </a> With new features such as workers, projects, variables, and versioned documentation alongside a slew of performance improvements and
-  <a href="https://github.com/PrefectHQ/prefect/blob/main/RELEASE-NOTES.md#release-2103" style="color: #07e798;">
+  <a href="https://github.com/PrefectHQ/prefect/blob/main/RELEASE-NOTES.md#release-2104" style="color: #07e798;">
+    <strong>Prefect 2.10.4 is here!</strong>
+  </a> With fixes and enhancements building on Prefect 2.10's new features like workers, projects, variables, and versioned documentation and
```
Comment (serinamarie) [inline docs/overrides/main.html:7]:
```suggestion
  </a> With fixes and enhancements building on Prefect 2.10's new features like workers, projects, variables, versioned documentation and
```

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,34 @@
 # Prefect Release Notes
 
+## Release 2.10.4
+
+This release further refines Prefect 2.10 with enhancements for [project deployments](https://docs.prefect.io/latest/concepts/projects/#the-deployment-yaml-file) and
+[workers](https://docs.prefect.io/latest/concepts/work-pools/#worker-overview), fixes for flow run cancellation and the worker CLI, and more.
+
+### More flexible project deployments
+Prior to this release, removing keys from a project's `deployment.yaml` caused an error. Thanks to the changes in [#9190](https://github.com/PrefectHQ/prefect/pull/9190), Prefect now uses default values for any required keys missing from 
+your project's configuration.
```
Comment (serinamarie) [inline RELEASE-NOTES.md:10]:
```suggestion
Prior to this release, removing keys from a project's `deployment.yaml` caused an error. Thanks to the changes in [#9190](https://github.com/PrefectHQ/prefect/pull/9190), Prefect now uses default values for any required keys missing from your project's configuration.
```

**Review by billpalombi — APPROVED** [review]
Thank you, lgtm!

**Pull request merged**